### PR TITLE
Feat/devx 405 abstract chainId

### PR DESCRIPTION
**Pull request by VGabriel45**
# Summary

The scope of this PR is to abstract away the need to provide a chain id manually, we can extract the chain id from bundler url as bundler is now required.

Related Issue: #DEVX-405

## Change Type

- [ ] Bug Fix
- [x] Refactor
- [x] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [ ] Other

# Checklist

- [x] My code follows this project's style guidelines
- [x] I've reviewed my own code
- [ ] I've added comments for any hard-to-understand areas
- [ ] I've updated the documentation if necessary
- [x] My changes generate no new warnings
- [x] I've added tests that prove my fix is effective or my feature works
- [x] All unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by VGabriel45** [comment]
The PR also contains latest "develop" branch updates

**Comment by joepegler** [comment]
Hmmm, I like this but I think I'd prefer to set the bundlerUrl (to our own bundler) only if the chainId is missing. So the chainId becomes mandatory if the bunderlUrl is missing, as the only dynamic part of our bundlerUrl is the chainId. 

I think removing the chainId from the config entirely could cause confusion as it's a param I'd expect to see. 

TLDR: Inferring the bunderUrl from the chainId would make more sense than inferring the chainId from the bundlerUrl.

File: packages/account/src/BaseSmartAccount.ts
Comment (AmanRaj1608) [inline packages/account/src/BaseSmartAccount.ts:1]: We don't need this in v4

**Review by AmanRaj1608 — CHANGES_REQUESTED** [review]
Do take a back merge from v4

**Comment by livingrockrises** [comment]
resolve conflicts
